ServerCore/Pages/Puzzles/Play.cshtml.cs

```diff
@@ -0,0 +1,247 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Puzzles
+{
+    /// <summary>
+    /// Model for the player's "Puzzles" page. Shows a list of the all the player puzzles (both team puzzles and single player puzzles).
+    /// </summary>
+    [Authorize()]
+    public class PlayModel : EventSpecificPageModel
+    {
+        // see https://docs.microsoft.com/en-us/aspnet/core/data/ef-rp/sort-filter-page?view=aspnetcore-2.1 to make this sortable!
+
+        public PlayModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        public IEnumerable<PuzzleView> VisibleTeamPuzzleViews { get; set; }
+
+        public IEnumerable<PuzzleView> VisibleSinglePlayerPuzzleViews { get; set; }
+
+        public PuzzleStateFilter? StateFilter { get; set; }
+
+        public SortOrder? TeamPuzzleSort { get; set; }
+
+        public SortOrder? SinglePlayerPuzzleSort { get; set; }
+
+        private const SortOrder DefaultSort = SortOrder.GroupAscending;
+
+        public bool ShowAnswers { get; set; }
+
+        public bool AllowFeedback { get; set; }
+
+        public Team Team { get; set; }
+
+        public async Task OnGetAsync(
+            SortOrder? teamPuzzleSort,
+            SortOrder? singlePlayerPuzzleSort,
+            PuzzleStateFilter? stateFilter)
+        {
+            TeamPuzzleSort = teamPuzzleSort;
+            SinglePlayerPuzzleSort = singlePlayerPuzzleSort;
+            StateFilter = stateFilter;
+            ShowAnswers = Event.AnswersAvailableBegin <= DateTime.UtcNow;
+            AllowFeedback = Event.AllowFeedback;
+
+            Dictionary<int, ContentFile> answers = await (from file in _context.ContentFiles
```

Comment:
Something seems wrong here - `answers` is ignored and then the computation of `answerFiles` is going to leave it empty because the `where` clause never gets answer files?

Reply:
This function does get answer files, but they don't get used for anything because line 240 only includes it if ShowAnswers is available

![image](https://github.com/PuzzleServer/mainpuzzleserver/assets/4121745/6fad3810-7fb3-458e-9a4b-ee513d351e19)

Reply:
That's not what I mean. Unless I am misreading the code, the local variable `answers` has the answers and does not appear to ever be used. The variable `answerFiles`, which is passed to `PopulatePuzzleViewsWithFiles` (the code you highlighted) in both cases, seems like it will always be empty, because it is built on line 71 but the `where` clause on line 61 never includes actual answer files. Am I misreading?

Reply:
AH. That was an unused variable that I left in there. Thanks for catching

Reply:
You've addressed half of this issue (`answers` being unused). The other half (`answerFiles` being always empty) should be addressed when you change:

`where file.Event == Event && file.FileType == ContentFileType.Puzzle`

to:

`where file.Event == Event && (file.FileType == ContentFileType.Puzzle || file.FileType == ContentFileType.Answer)`

(unless I am misreading something)

Reply:
omg. You are right. What would we do without you. Thanks again!